cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexUpdaterTest.java

```diff
@@ -125,6 +127,123 @@ public void updateOneFieldIndex() throws Exception {
         }
     }
 
+
+    @Test
+    public void updateOneFieldIndexMultithreaded() throws Exception {
+        createIndex("basic", Arrays.<Object>asList("name"));
+
+        assertThat(getIndexSequenceNumber("basic"), is(0l));
+
+        String table = IndexManager.tableNameForIndex("basic");
+        String sql = String.format("SELECT * FROM %s", table);
+        Cursor cursor = null;
+        try {
+            SQLDatabase db = TestUtils.getDatabaseConnectionToExistingDb(this.db);
+            cursor = db.rawQuery(sql, new String[]{});
+            assertThat(cursor.getCount(), is(0));
+        } catch (SQLException e) {
+            Assert.fail(String.format("SQLException occurred executing %s: %s", sql, e));
+        } finally {
+            DatabaseUtils.closeCursorQuietly(cursor);
+        }
+
+        final int nDocs = 50;
+        final int nThreads = 8;
+        final int nUpdates = 10; // update index after how many docs?
+
+        // populate nDocs documents per thread, across nThreads simultaneously
+        // and update index after each insert
```

Comment:
@tomblench did you miss this note? I think it needs to say "and update index after nUpdates" now